src/db/couchDevices.ts

```diff
@@ -176,9 +178,9 @@ export const countDevicesByLocation = async (
   })
 
   return results.rows.map(row => {
-    const key: string[] = asArray(asString)(row.key)
+    const key = asEither(asArray(asString), asNull)(row.key)
```

Comment:
How about `asOptional(asArray(asString), [])`? That will handle `null` and do the fallback in one thing. But this is fine if there is some reason for it.

Reply:
Okay, I'll test with that solution before merging.